avatars/avatars.py

```diff
@@ -107,7 +108,16 @@ def get_data(self, name, email_or_id, kind="user"):
         # so use the username in that case.
         username_email_or_id = email_or_id or name
         username_email_or_id = Bytes.for_string_or_unicode(username_email_or_id).as_unicode()
-        hash_value = hashlib.md5(username_email_or_id.strip().lower().encode("utf-8")).hexdigest()
+
+        # If fips mode is enabled, we must use 256. In future Quay versions, this should be the default. Flag is set for backwards compatibility.
+        if features.FIPS:
+            hash_value = hashlib.sha256(
+                username_email_or_id.strip().lower().encode("utf-8")
+            ).hexdigest()
+        else:
+            hash_value = hashlib.md5(
```

Comment:
Can probably move this into GravatarAvatar specific implentation, since it's the only one that needs to use md5: https://en.gravatar.com/site/implement/hash/

Reply:
I'm a little worried about the size of that refactor, as `get_data` is used by a few publicly used methods (`get_data_for_external_user`, `get_data_for_org`, `get_data_for_team`, `get_data_for_user`).  

Perhaps we could accept this PR to solve the ticket related to FIPS and open a feature request to think about refactoring `GravatarAvatar` at a later time?

Also I'd love to help contribute anyway I can.  I'm particularly excited about getting this ticket merged in, so anything I can do to help move this along I'd be more than happy to help!